ls1intum/Artemis
# `Programming exercises`: Synchronize result display in exercise view

### Checklist
#### General


#### Client

### Motivation and Context


Currently the result history and the result string for the newest result used different stylings for the icons and text color.

### Description

This PR unifies these two views and refactors the used methods to reduce code duplication.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming Exercise

1. Log in to Artemis
2. Navigate to the exercise and participate as a student
3. Check that in different configurations (befor/after/without due date, different assessment types, ...) and verify that the results are styled the same.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| result.utils.ts | 72% | 87% |


### Screenshots


Before (Red Cross vs. question mark)
![Bildschirmfoto 2022-10-24 um 16 28 08](https://user-images.githubusercontent.com/38322605/197550617-6403c389-c113-4aa1-82f3-172e97702e12.png)
After (Both question marks)
![Bildschirmfoto 2022-10-24 um 16 19 17](https://user-images.githubusercontent.com/38322605/197550853-3e9e8fbc-3811-48b2-a777-37600ece57c5.png)
